### PR TITLE
Test scons multithreaded random build

### PR DESCRIPTION
**Pull request by maxime-desroches**

### CODE REVIEW

**Comment by adeebshihadeh** [comment]
Can we randomize the build order? That would catch this on the slow CI runners.

**Comment by adeebshihadeh** [comment]
Looks like there's a `--random` flag.

**Comment by sshane** [comment]
yeah there's more than just this:

```python
cd /home/batman/openpilot/selfdrive/controls/lib/lateral_mpc_lib && python3 lat_mpc.py
Traceback (most recent call last):
  File "/home/batman/openpilot/selfdrive/controls/lib/lateral_mpc_lib/lat_mpc.py", line 6, in <module>
    from common.realtime import sec_since_boot
  File "/home/batman/openpilot/common/realtime.py", line 10, in <module>
    from common.clock import sec_since_boot  # pylint: disable=no-name-in-module, import-error
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ModuleNotFoundError: No module named 'common.clock'

Traceback (most recent call last):
  File "/home/batman/openpilot/tools/cabana/dbc/generate_dbc_json.py", line 5, in <module>
    from selfdrive.car.car_helpers import get_interface_attr
  File "/home/batman/openpilot/selfdrive/car/car_helpers.py", line 5, in <module>
    from common.params import Params
  File "/home/batman/openpilot/common/params.py", line 1, in <module>
    from common.params_pyx import Params, ParamKeyType, UnknownKeyName, put_nonblocking, put_bool_nonblocking # pylint: disable=no-name-in-module, import-error
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ModuleNotFoundError: No module named 'common.params_pyx'
```

**Comment by adeebshihadeh** [comment]
@maxime-desroches let's first write a good test that runs quickly on your workstation (with lots of cores). then we can run it in jenkins, which is similar to your workstation

**Comment by maxime-desroches** [comment]
It catch some error

File: Jenkinsfile [hunk 1]
```diff
@@ -128,6 +128,22 @@ pipeline {
...
+              filename 'Dockerfile.openpilot_base'
+              args '--user=root'
+            }
+          }
+          steps {
+            sh "git config --global --add safe.directory '*'"
+            sh "git submodule update --init --depth=1 --recursive"
+            sh "git lfs pull"
+            sh "scons --clean && scons --no-cache --random -j42"
+            sh "scons --clean && scons --no-cache --random -j42"
```
Comment (adeebshihadeh) [inline Jenkinsfile:143]: how about we do two passes: one normal build with lots of cores and add random to the second one?

**Review by adeebshihadeh — CHANGES_REQUESTED** [review]
Let's make that change and fix the errors

**Comment by maxime-desroches** [comment]
Ran over 35 times, that's all the bugs it could find

File: Jenkinsfile [hunk 2]
```diff
@@ -128,6 +128,29 @@ pipeline {
...
+          agent {
+            dockerfile {
+              filename 'Dockerfile.openpilot_base'
+              args '--user=root'
+            }
+          }
+          steps {
+            sh "git config --global --add safe.directory '*'"
+            sh "git submodule update --init --depth=1 --recursive"
+            sh "git lfs pull"
```
Comment (adeebshihadeh) [inline Jenkinsfile:141]: do we need lfs for the build?

File: selfdrive/controls/lib/lateral_mpc_lib/SConscript [hunk 3]
```diff
@@ -46,7 +46,8 @@ generated_files = [
 acados_dir = '#third_party/acados'
 acados_templates_dir = '#third_party/acados/acados_template/c_templates_tera'
 
-source_list = ['lat_mpc.py',
+source_lat = 'lat_mpc.py'
```
Comment (adeebshihadeh) [inline selfdrive/controls/lib/lateral_mpc_lib/SConscript:49]: don't think this needs an intermediate variable

File: selfdrive/controls/lib/longitudinal_mpc_lib/SConscript [hunk 4]
```diff
@@ -53,7 +53,8 @@ generated_files = [
 acados_dir = '#third_party/acados'
 acados_templates_dir = '#third_party/acados/acados_template/c_templates_tera'
 
-source_list = ['long_mpc.py',
+source_long = 'long_mpc.py'
```
Comment (adeebshihadeh) [inline selfdrive/controls/lib/longitudinal_mpc_lib/SConscript:56]: same here

**Review by adeebshihadeh — APPROVED** [review]

**Pull request merged**